### PR TITLE
Add tests for writeStorageObject

### PR DESCRIPTION
**Pull request by aminalaee**
This PR adds:
- tests for writeStorageObject
- fix session missing from writeStorageObject grpc

### CODE REVIEW

File: lib/src/nakama_client/nakama_grpc_client.dart [hunk 1]
```diff
@@ -326,26 +326,25 @@ class NakamaGrpcClient extends NakamaBaseClient {
...
-              : null,
-          permissionRead: readPermission != null
-              ? Int32Value(
-                  value: StorageReadPermission.values.indexOf(readPermission),
-                )
-              : null,
-        ),
-      ],
-    ));
+    return _client.writeStorageObjects(
```
Comment (aminalaee) [inline lib/src/nakama_client/nakama_grpc_client.dart:329]: I also noticed writeStorageObject could have been async, but it's the only method not using the async
Is there a specific reason or a an issue? Doesn't it block the event loop to run sync?

**Comment by aminalaee** [comment]
@obrunsmann Any chance to review these?

File: lib/src/nakama_client/nakama_grpc_client.dart (hunk 1 above)
Comment (obrunsmann) [reply lib/src/nakama_client/nakama_grpc_client.dart:328]: What do you mean with that? From the `writeStorageObject` we are returning a Future, so this method is already async 🤔

Comment (obrunsmann) [reply lib/src/nakama_client/nakama_grpc_client.dart:328]: Or do you mean this async keyboard here?

```dart
  @override
  Future<void> writeStorageObject({
    required model.Session session,
    String? collection,
    String? key,
    String? value,
    String? version,
    StorageWritePermission? writePermission,
    StorageReadPermission? readPermission,
  }) >>async<< {
```

**Review by obrunsmann — APPROVED** [review]
LGTM, thanks for contribution!

**Comment by aminalaee** [comment]
Yes @obrunsmann I mean the async keyword is missing and because of that the call to the backend is not awaited.

**Comment by obrunsmann** [comment]
> Yes @obrunsmann I mean the async keyword is missing and because of that the call to the backend is not awaited.

This is part of the caller, there is no difference between the following:

```dart
// A
Future<void> doSomething() {
  return _someNakamaTask();
}

// B
Future<void> doSomething() async {
  return _await someNakamaTask();
}
```

Both codes doing absolutely the same, the caller has to await the execution:

```dart
void main() async {
  print('Hello (:');
  await doSomething();
}
```

The difference between A and B is just that in A we return the nakamasTask's future and in B we are creating our own.

**Pull request merged**